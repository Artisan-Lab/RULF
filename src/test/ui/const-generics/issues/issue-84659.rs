#![allow(incomplete_features)]
#![feature(generic_const_exprs)]

trait Bar<const N: usize> {}

trait Foo<'a> {
    const N: usize;
    type Baz: Bar<{ Self::N }>;
    //~^ ERROR: unconstrained generic constant
}

fn main() {}
