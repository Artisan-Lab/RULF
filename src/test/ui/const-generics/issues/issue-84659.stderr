error: unconstrained generic constant
  --> $DIR/issue-84659.rs:8:15
   |
LL |     type Baz: Bar<{ Self::N }>;
   |               ^^^^^^^^^^^^^^^^
   |
   = help: try adding a `where` bound using this expression: `where [(); { Self::N }]:`

error: aborting due to previous error

